src/goose.rs

```diff
@@ -1222,6 +1243,28 @@ impl GooseClient {
                     }
                     raw_request.set_status_code(Some(status_code));
                     raw_request.set_final_url(r.url().as_str());
+
+                    // Load test client was redirected.
+                    if raw_request.url != raw_request.final_url {
+                        // Extract the base_url of the request made.
+                        if let Ok(original_url) = Url::parse(&raw_request.url) {
+                            if let Some(original_host) = original_url.host() {
+                                if let Some(test_base_url) = self.get_base_url().await.host() {
+                                    // Check if the base_url of the request made matches the base_url load test.
```

Comment:
I would just check and store if it was a relative request.

Reply:
We don't actually know here if the URL is relative or not, as in a previous step we prepend the load test host. We could add an extra field to track if a URL was relative or absolute, but Rust's built in string handling makes it pretty easy to just check the first part.